### PR TITLE
Allow IPv4 address substitutions in EBS volume names

### LINKED ISSUE #488: Add support for variable expansion from the user data
To the best of my knowledge, the user data is static and can be found under /etc/taupage.yaml

Some use cases could benefit from having some basic variable expansion, based on a limited, well-known set of variables.

For [PlanB Cassandra](https://github.com/zalando-stups/planb-cassandra) each node has its own unique User Data but the only difference is the EBS volume name, for example:
```
volumes:
  ebs: {/dev/xvdf: cassandra-172.31.128.40}
```

Some tools that perform EC2 orchestration would benefit if the User Data was just a template and each instance would expand the known variables. For the above example, this would be something like:
```
volumes:
  ebs: {/dev/xvdf: cassandra-${local-ipv4}}
```

If this sounds useful, I propose an initial implementation with the following variables, directly from the metadata service - http://169.254.169.254/latest/meta-data/:
* local-ipv4
* hostname
* public-ipv4
* public-hostname

This would go into the get_config() method in https://github.com/zalando-stups/taupage/blob/master/runtime/usr/local/lib/python3.4/dist-packages/taupage/__init__.py providing the expansion for all calls already depending on this method. Any other requirements, like having access to the raw file, could still read the file /etc/taupage.yaml.

### PR DESCRIPTION
**Pull request by a1exsh**
Fixes #488

Limit scope of substitutions to EBS volume names only.  This is not needed in
any other places of configuration (otherwise there would be open issues for
those as well).

Only allow `local-ipv4` and `public-ipv4` vars from instance metadata: for the
volume name substitutions these are the most useful ones.

### CODE REVIEW

File: runtime/opt/taupage/init.d/10-prepare-disks.py [hunk 1]
```diff
@@ -30,6 +31,29 @@ def zone():
...
+        k: d[k]
+        for k in keys
+        if k in d
+    }
+
+
+def allowed_volume_name_subs():
+    """Helper to return the allowed volume name substitutions from the instance metadata"""
+    return select_keys(
+        boto.utils.get_instance_metadata(),
```
Comment (aermakov-zalando) [inline runtime/opt/taupage/init.d/10-prepare-disks.py]: IMDS has hiccups sometimes. Does this retry?

Comment (aermakov-zalando) [reply runtime/opt/taupage/init.d/10-prepare-disks.py]: Although I guess it's used in other parts of the script anyway, so it won't be worse.

**Comment by aermakov-zalando** [comment]
👍

File: runtime/opt/taupage/init.d/10-prepare-disks.py (hunk 1 above)
Comment (a1exsh) [reply runtime/opt/taupage/init.d/10-prepare-disks.py]: I've also noticed that `find_volume` isn't annotated with `@retry`.  Should we add it to this PR or make a separate one just for that?

Comment (aermakov-zalando) [reply runtime/opt/taupage/init.d/10-prepare-disks.py]: Let's do it separately.

**Comment by jrake-revelant** [comment]
👍

**Comment by a1exsh** [comment]
Hm, I've just tried the actual `${local-ipv4}` substitution and it doesn't work because of the `-` in the middle.  Another PR will be needed to make it work.

**Pull request merged**